### ISSUE TITLE
Lint when `>` or `>=` is used i.e. compared values are not listed in (number line) order

### ISSUE THREAD

**Issue by Skgland** [rust-lang/rust-clippy#13082]
### What it does

Check that `<` and `<=` are used rather than `>` or `>=` so that compared values are listed in order.

### Advantage

- unified style 
- some find it easier to reason about (prompted by https://github.com/rust-lang/cargo/pull/14221#discussion_r1672445595)

### Drawbacks

- Its rather opinionated so should probably be allow by default style/pedantic lint
- inversion might not be possible if only the `PartialEq` impl for one direction is implemented, so it might need to be limited to cases where the operands implement `Eq`

### Example

Could be written as:

```rust
if val >= min && max >= val {
    // do something
}
```

```rust
if min <= val && val <= max {
    // do something
}
```

Preferably it would also recognize common expressions so that 

```rust
if val <= max && min <= val {
    // do something
}
if max <= val || val <= min {
    // do something
}
```

would become 

```rust
if min <= val && val <= max {
    // do something
}
if val <= min && max <= val {
    // do something
}
```

i.e. for `&&` move common value to the operator and for `||` move common value away from the operator
```
